### PR TITLE
[Fix] UnmockOptions implements IUnmockOptions

### PR DESCRIPTION
**Pull request by idantene**
Also adds a basic composition test

### CODE REVIEW

File: packages/unmock-core/src/__tests__/composition.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import {
```
Comment (mikesol) [inline packages/unmock-core/src/__tests__/composition.test.ts]: Check out https://jestjs.io/docs/en/es6-class-mocks - it is a cleaner way to mock classes.

Comment (idantene) [reply packages/unmock-core/src/__tests__/composition.test.ts]: Not sure how these would work w/o a class; can't find a way to use an interface cleanly with `jest.mock` in a way that TS won't complain.

Comment (idantene) [reply packages/unmock-core/src/__tests__/composition.test.ts]: Or did you mean just using `jest.fn()` in all pseudo-functions?

**Comment by mikesol** [comment]
Lemme dig in, one sec.

**Comment by mikesol** [comment]
See the most recent commit.

**Comment by idantene** [comment]
> See the most recent commit.

:man_facepalming: my bad.

**Comment by idantene** [comment]
A weird error in CircleCI though (can't reproduce locally) -
`    ENOMEM: not enough memory, read`

**Pull request merged**